mysql/resource_user.go

```diff
@@ -2,15 +2,16 @@ package mysql
 
```

Comment:
For tests:

implement your own check similar to this one
https://github.com/petoju/terraform-provider-mysql/blob/master/mysql/resource_user_test.go#L151-L181
that will check, whether one can authenticate as given user with given password. Maybe your test will just try to connect to MySQL with that data.

Then implement test like this:
https://github.com/petoju/terraform-provider-mysql/blob/master/mysql/resource_user_test.go#L54-L89
you don't have to run TestCheckResourceAttr. Without that, you'll run some written config, then possibly do some check. Next step will change password by specifying another config and run the check again to verify the old password stayed in place.
The last step would be to check, whether authentication with some random password works - that is a verification whether our tests make sense.

I also recommend doing test for the case one doesn't want to retain password.